### PR TITLE
Fixes #1737 - Create checkbox component

### LINKED ISSUE #1737: Create clay-checkbox component for v3.0.0

### PR DESCRIPTION
**Pull request by bryceosterhaus**

### CODE REVIEW

File: packages/clay-checkbox/src/ClayCheckbox.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+					{...otherProps}
+					checked={checked}
+					className={classNames('custom-control-input', className)}
+					ref={inputRef}
+					type="checkbox"
+					value={checked.toString()}
+				/>
+
+				{label && (
+					<span className="custom-control-label">
```
Comment (matuzalemsteles) [inline packages/clay-checkbox/src/ClayCheckbox.tsx]: This should be out of the condition, just your content, we should also add an API or use the children so we can add other content that is not text, we will be using it in a ClayCard for example.

See https://github.com/liferay/clay/blob/2.x-develop/packages/clay-checkbox/src/ClayCheckbox.soy#L93-L108

File: packages/clay-checkbox/src/ClayCheckbox.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+						<span className="custom-control-label-text">
```
Comment (matuzalemsteles) [inline packages/clay-checkbox/src/ClayCheckbox.tsx]: At some point we will need to add the `sr-only` class. Previously we had `showLabel` API for this.

Comment (bryceosterhaus) [reply packages/clay-checkbox/src/ClayCheckbox.tsx]: I talked to @pat270 about this and we settled on removing `sr-only` and having people rely on using aria attributes.

File: packages/clay-checkbox/src/ClayCheckbox.tsx (hunk 1 above)
Comment (bryceosterhaus) [reply packages/clay-checkbox/src/ClayCheckbox.tsx]: I was trying to avoid using multiple inter-dependent props such as `showLabel`, `label`, and `children`. By using just label, we can assure its the only place to add content. One way to allow for more than text would be to check the type of the label and conditionally render the class names. What do you think?

File: packages/clay-checkbox/src/ClayCheckbox.tsx (hunk 2 above)
Comment (matuzalemsteles) [reply packages/clay-checkbox/src/ClayCheckbox.tsx]: Hmm, we have to be careful about that. We have to add this information in the migration guide and also add as a good practice in the new documentation.

File: packages/clay-checkbox/src/ClayCheckbox.tsx (hunk 1 above)
Comment (matuzalemsteles) [reply packages/clay-checkbox/src/ClayCheckbox.tsx]: Hmm, just be concerned that the API name is not self explanatory for this use case. But it sounds good to me to.

Comment (bryceosterhaus) [reply packages/clay-checkbox/src/ClayCheckbox.tsx]: Ill add it right now and test it out. Maybe be some edge case where a user wants both text and markup. Ill update the pr and let you know

**Comment by bryceosterhaus** [comment]
@matuzalemsteles updated. I included a aria-label example in the demo. Once the `v2-v3.md` file is in, I will also update that doc.

**Comment by matuzalemsteles** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: packages/clay-checkbox/src/ClayCheckbox.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+				/>
+
+				<span className="custom-control-label">
+					{label && (
+						<span className="custom-control-label-text">
+							{label}
+						</span>
+					)}
+
+					{children}
```
Comment (matuzalemsteles) [inline packages/clay-checkbox/src/ClayCheckbox.tsx]: By markup it is expected that the content stays outside the `span`.

**Comment by bryceosterhaus** [comment]
@matuzalemsteles updated and moved markup outside of span

**Comment by bryceosterhaus** [comment]
@matuzalemsteles updated and fixed conflicts

**Comment by matuzalemsteles** [comment]
Merged!
